File: server/controllers/LocationsController.js
const { Location } = require('../models/LocationModel.js');

module.exports.getAllLocations = async (req, res) => {
    try{
        const locations = await Location.findAll();
        return res.status(200).json(locations)
    } catch(err) {
        return res.status(400).json({ msg: 'Error retrieving all locations'})
    }
}

module.exports.getLocationById = async (req, res) => {
    const { id } = req.params;
    try {
        const location = await Location.findOne({where: {id: id}});
        return res.status(200).json(location)
    } catch (err) {
        return res.status(400).json({ msg: 'Could not find location', err})
    }
}

File: server/app/controllers/locations_controller.rb
class LocationsController < ApplicationController
    skip_before_action :authorized, only: [:index, :show]
    # GET /locations
    def index
        location = Location.all
        render json: location, status: :ok
    end
    # GET /locations/:id
    def show 
        location = Location.find_by(id: params[:id])
        render json: location, status: :ok
    end
end
